### PR TITLE
Added "enable OpenShift" boolean flag to OpenShift daemon

### PR DESCRIPTION
**Pull request by miminar**
This starts AE enablement work within OpenShift.

The boolean will enable OpenShift Enterprise features like S2I builds and UI at runtime.

It will also affect console output (help messages won't refer to disabled features).

I believe it makes sense to make the flag hidden. WDYT?

### CODE REVIEW

File: pkg/cmd/server/start/master_args.go [hunk 1]
```diff
@@ -40,8 +40,8 @@ type MasterArgs struct {
 	// with any IP ranges assigned to nodes for pods.
 	PortalNet flagtypes.IPNet
 
-	// MasterPublicAddr is the master address for use by public clients, if different (host, host:port,
```
Comment (ncdc) [inline pkg/cmd/server/start/master_args.go:43]: Why did you remove this comment?

File: pkg/cmd/server/start/master_args.go [hunk 2]
```diff
@@ -75,6 +75,7 @@ type MasterArgs struct {
 func BindMasterArgs(args *MasterArgs, flags *pflag.FlagSet, prefix string) {
 	flags.Var(&args.MasterAddr, prefix+"master", "The master address for use by OpenShift components (host, host:port, or URL). Scheme and port default to the --listen scheme and port. When unset, attempt to use the first public IPv4 non-loopback address registered on this host.")
 	flags.Var(&args.MasterPublicAddr, prefix+"public-master", "The master address for use by public clients, if different (host, host:port, or URL). Defaults to same as --master.")
+	flags.BoolVar(&args.OpenshiftEnabled, prefix+"openshift-enabled", false, "Controls if openshift additions are enabled or not.")
```
Comment (ncdc) [inline pkg/cmd/server/start/master_args.go]: Controls if **OpenShift** additions

**Comment by miminar** [comment]
Thanks @ncdc, I've resubmitted against upstream with some corrections. See [PR#3896](https://github.com/openshift/origin/pull/3896).

File: pkg/cmd/server/start/master_args.go (hunk 1 above)
Comment (miminar) [inline pkg/cmd/server/start/master_args.go:43]: @ashcrow ^^
I've re-added original comment. Is that all right?
